plutus-core/plutus-ir/Language/PlutusIR/TypeCheck/Internal.hs

```diff
@@ -104,8 +100,8 @@ type PirTCEnv uni e a = TypeCheckM uni (PirTCConfig uni) e a
 -- See the [Global uniqueness] and [Type rules] notes.
 -- | Check a 'Term' against a 'NormalizedType'.
 checkTypeM
-    :: (GShow uni, GEq uni, DefaultUni <: uni, AsTypeErrorExt e uni ann, AsTypeError e (Term TyName Name uni ()) uni ann)
-    => ann -> Term TyName Name uni ann -> Normalized (Type TyName uni ()) -> PirTCEnv uni e ()
+    :: (GShow uni, GEq uni, Ix fun, AsTypeErrorExt e uni ann, AsTypeError e (Term TyName Name uni fun ()) uni fun ann)
```

Comment:
what role is `Ix` playing here?

Reply:
We need it for type checking to look up the precomputed type of a built-in function in an `Array`.